bundles/org.openhab.binding.juicenet/src/main/java/org/openhab/binding/juicenet/internal/handler/JuiceNetDeviceHandler.java

```diff
@@ -0,0 +1,295 @@
+import org.openhab.core.library.types.DateTimeType;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.library.unit.ImperialUnits;
+import org.openhab.core.library.unit.SIUnits;
+import org.openhab.core.library.unit.Units;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.thing.binding.BridgeHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link JuiceNetDeviceHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Jeff James - Initial contribution
+ */
+@NonNullByDefault
+public class JuiceNetDeviceHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(JuiceNetDeviceHandler.class);
+
+    protected String name = "";
+    protected String id = "";
+    protected String token = "";
+
+    protected long targetTimeTou = 0;
+    protected long lastInfoTimestamp = 0;
+
+    JuiceNetApi.JuiceNetApiDeviceStatus deviceStatus = new JuiceNetApi.JuiceNetApiDeviceStatus();
+    JuiceNetApi.JuiceNetApiInfo deviceInfo = new JuiceNetApi.JuiceNetApiInfo();
+    JuiceNetApi.JuiceNetApiTouSchedule deviceTouSchedule = new JuiceNetApi.JuiceNetApiTouSchedule();
+    JuiceNetApi.JuiceNetApiCar deviceCar = new JuiceNetApi.JuiceNetApiCar();
+
+    // This holds the last time the info was updated and is compared everytime we get device status to determine
+    // if the deviceInfo should be updated. Set the time in the past to ensure info is updated on initial call.
+    protected ZonedDateTime infoTimestamp = ZonedDateTime.of(2000, 1, 1, 0, 0, 0, 0, ZoneOffset.UTC);
+
+    public JuiceNetDeviceHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        logger.trace("JuiceNetDeviceHandler:initialize");
+        Map<String, String> properties = editProperties();
+
+        name = Objects.requireNonNull(properties.get(DEVICE_PROP_NAME));
+        id = Objects.requireNonNull(properties.get(DEVICE_PROP_UNIT_ID));
+        token = Objects.requireNonNull(properties.get(DEVICE_PROP_TOKEN));
+
+        updateStatus(ThingStatus.UNKNOWN);
+
+        // Example for background initialization:
+        scheduler.execute(() -> {
+            try {
+                queryDeviceStatusAndInfo();
+            } catch (JuiceNetApiException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR, e.toString());
```

Comment:
This exception is probably thrown in case of network/API issues? In this case it should probably be **COMMUNICATION_ERROR**, not **CONFIGURATION_ERROR**.